src/cursor/abstract_cursor.ts

```diff
@@ -778,62 +778,51 @@ function cleanupCursor(
   const error = options?.error;
   const needsToEmitClosed = options?.needsToEmitClosed ?? cursor[kDocuments].length === 0;
```

Comment:
I don't object to the refactor but

1. why not just return early if `cursor[kKilled]`
2. why cleanup if `cursor[kClosed]`?  with your changes, `cursor[kClosed]` is set after the cleanup

couldn't we just add this at the top of complete cleanup:

```
if (cursor[kKilled] || cursor[kClosed]) {
  process.nextTick(callback);
  return;
}
```

Reply:
Good question

Reply:
I've changed the function to look like this, there's tests that fail though, looking into if the behavior they are looking for is something we need to be concerned about.